### ISSUE TITLE
Condition Error in the 'evaluate' function.

### ISSUE THREAD

**Issue by cifang-Kasit** [wenhaomin/DiffSTG#4]
![image](https://github.com/wenhaomin/DiffSTG/assets/74644912/c6c2b195-f8dd-4587-87de-b111f48f0d76)

Dear author, 

Thank you for bring us the nice work.

As the picture shown, in the evaluate function, the way that extract the information from the condition will lead to the information loss of the 'pos_w'. 

Would you mind take a quick look and see if this is an issue or my misunderstand?

Thank you so much.

**Comment by wenhaomin** [comment]
Hi, thank you for the question.  Currently, we do not use the pos_w.  We leave the interface here, feel free to use them make your model design!

**Comment by cifang-Kasit** [comment]
Thank you for your quick reply. 
I used the hyperparameter that provided by the code with the PEMS08 Dataset.

'T_h': 12,
 'T_p': 12,
 'batch_size': 8,
 'data': {'feature_file': 'c:\\Users\\dell\\Documents\\GPU_directories\\Ten\\DiffSTG-main\\DiffSTG-main/data/dataset/PEMS08/flow.npy',
          'name': 'PEMS08',
          'num_features': 1,
          'num_recent': 1,
          'num_vertices': 170,
          'path': 'c:\\Users\\dell\\Documents\\GPU_directories\\Ten\\DiffSTG-main\\DiffSTG-main/data/dataset/',
          'points_per_hour': 12,
          'spatial': 'c:\\Users\\dell\\Documents\\GPU_directories\\Ten\\DiffSTG-main\\DiffSTG-main/data/dataset/PEMS08/adj.npy',
          'test_start_idx': 14284,
          'val_start_idx': 10713},
 'device': device(type='cuda', index=1),
 'early_stop': 10,
 'epoch': 300,
 'gpu_id': 1,
 'is_test': False,
 'load_model': True,
 'load_model_path': './DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg',
 'log_path': 'c:\\Users\\dell\\Documents\\GPU_directories\\Ten\\DiffSTG-main\\DiffSTG-main/output/log//UGnet+32+200+quad+0.1+200+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8+True+./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg.log',
 'logger': <utils.common_utils.Logger object at 0x00000152396EE520>,
 'lr': 0.002,
 'mask_ratio': 0.0,
 'model': {'C': 32,
           'F': 1,
           'N': 200,
           'T_h': 12,
           'T_p': 12,
           'V': 170,
           'beta_end': 0.1,
           'beta_schedule': 'quad',
           'channel_multipliers': [1, 2],
           'd_h': 32,
           'day_len': 288,
           'device': device(type='cuda', index=1),
           'epsilon_theta': 'UGnet',
           'is_label_condition': True,
           'n_channels': 32,
           'sample_steps': 200,
           'sample_strategy': 'ddim_multi',
           'supports_len': 2,
           'week_len': 7},
 'model_name': 'DiffSTG',
 'n_samples': 8,
 'nni': False,
 'optimizer': 'adam',
 'start_epoch': 0,
 'trial_name': 'UGnet+32+200+quad+0.1+200+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8+True+./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg',
 'wd': 1e-05}
model loaded from: ./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg
sample num: 10690
sample num: 3548
sample num: 3549

Sorry for directly plot the log. However, I can not achieve the performance as the paper presented.
![image](https://github.com/wenhaomin/DiffSTG/assets/74644912/945dcdc6-3173-43df-9b1c-6d600c21d839)


May I kindly ask some suggestion on the hyperparameter that you performed? For example, I am wondering if the early step is too small. 

Since the training really takes time, may I ask if the weight is possible to share or if it's possible to achieve the similar performance from my device.

**Comment by poppy148** [comment]
Hello, I encountered a similar issue while trying to reproduce the results from the paper. I set the early stop to 5 steps and experimented with multiple sets of hyperparameters, but the experimental results are significantly different from those reported in the paper. The inference process is also time-consuming. If you have successfully reproduced the results from the paper, could you kindly share the hyperparameter settings suitable for the PEMS08 dataset? I would greatly appreciate it if you could provide those.

I am using an RTX 2080ti GPU, and the experimental results I obtained are as follows (with early stop uniformly set to 5):

DiffSTG_PEMS08_12_12_bs24_lr0.006_beta0.1_ddpmstep100_ddimstep40: mae: 56.2306, rmse: 70.6376
DiffSTG_PEMS08_12_12_bs24_lr0.006_beta0.1_ddpmstep200_ddimstep40: mae: 31.1109, rmse: 43.5880
DiffSTG_PEMS08_12_12_bs8_lr0.002_beta0.1_ddpmstep200_ddimstep40: mae: 52.7865, rmse: 70.1869
DiffSTG_PEMS08_12_12_bs16_lr0.005_beta0.1_steps200_ddimstep40: mae: 45.7245, rmse: 59.2612
DiffSTG_PEMS08_12_12_bs16_lr0.005_beta0.2_steps100_ddimstep40: mae: 34.9530, rmse: 48.8223
DiffSTG_PEMS08_12_12_bs24_lr0.008_beta0.2_steps100_ddimstep40: mae: 35.2182, rmse: 47.5447
I am eagerly awaiting your and the authors' response and help. Thank you!

**Comment by yejinxueshi** [comment]
> Thank you for your quick reply. I used the hyperparameter that provided by the code with the PEMS08 Dataset.
> 
> 'T_h': 12, 'T_p': 12, 'batch_size': 8, 'data': {'feature_file': 'c:\Users\dell\Documents\GPU_directories\Ten\DiffSTG-main\DiffSTG-main/data/dataset/PEMS08/flow.npy', 'name': 'PEMS08', 'num_features': 1, 'num_recent': 1, 'num_vertices': 170, 'path': 'c:\Users\dell\Documents\GPU_directories\Ten\DiffSTG-main\DiffSTG-main/data/dataset/', 'points_per_hour': 12, 'spatial': 'c:\Users\dell\Documents\GPU_directories\Ten\DiffSTG-main\DiffSTG-main/data/dataset/PEMS08/adj.npy', 'test_start_idx': 14284, 'val_start_idx': 10713}, 'device': device(type='cuda', index=1), 'early_stop': 10, 'epoch': 300, 'gpu_id': 1, 'is_test': False, 'load_model': True, 'load_model_path': './DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg', 'log_path': 'c:\Users\dell\Documents\GPU_directories\Ten\DiffSTG-main\DiffSTG-main/output/log//UGnet+32+200+quad+0.1+200+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8+True+./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg.log', 'logger': <utils.common_utils.Logger object at 0x00000152396EE520>, 'lr': 0.002, 'mask_ratio': 0.0, 'model': {'C': 32, 'F': 1, 'N': 200, 'T_h': 12, 'T_p': 12, 'V': 170, 'beta_end': 0.1, 'beta_schedule': 'quad', 'channel_multipliers': [1, 2], 'd_h': 32, 'day_len': 288, 'device': device(type='cuda', index=1), 'epsilon_theta': 'UGnet', 'is_label_condition': True, 'n_channels': 32, 'sample_steps': 200, 'sample_strategy': 'ddim_multi', 'supports_len': 2, 'week_len': 7}, 'model_name': 'DiffSTG', 'n_samples': 8, 'nni': False, 'optimizer': 'adam', 'start_epoch': 0, 'trial_name': 'UGnet+32+200+quad+0.1+200+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8+True+./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg', 'wd': 1e-05} model loaded from: ./DiffSTG-main/output/model/UGnet+32+100+quad+0.1+100+ddim_multi+12+8+True+PEMS08+0.0+False+False+0.002+8N-100+T_h-12+T_p-12+epsilon_theta-UGnet.dm4stg sample num: 10690 sample num: 3548 sample num: 3549
> 
> Sorry for directly plot the log. However, I can not achieve the performance as the paper presented. ![image](https://private-user-images.githubusercontent.com/74644912/339787392-945dcdc6-3173-43df-9b1c-6d600c21d839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0ODAwMzUsIm5iZiI6MTczMTQ3OTczNSwicGF0aCI6Ii83NDY0NDkxMi8zMzk3ODczOTItOTQ1ZGNkYzYtMzE3My00M2RmLTliMWMtNmQ2MDBjMjFkODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDA2MzUzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlYTM4YjZhZDA2OTM0YTFlNjAxMjhiMDhlNzQ5ZGFhODk1MTk0ZWVmODkzYzAxZjZiODRmYjhhYjQ0MTEwMzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tB8jelkoOcckpduS_5NaFeUNhKbM-x5XxC6sQsexQo8)
> 
> May I kindly ask some suggestion on the hyperparameter that you performed? For example, I am wondering if the early step is too small.
> 
> Since the training really takes time, may I ask if the weight is possible to share or if it's possible to achieve the similar performance from my device.

Have you solved this? I can't reproduce the results in the paper
